samples/nrf9160/modem_shell/src/shell.c

```diff
@@ -181,6 +181,19 @@ int lwm2m_carrier_event_handler(const lwm2m_carrier_event_t *event)
 }
 #endif
 
+int sleep_shell(const struct shell *shell, size_t argc, char **argv)
+{
+	long sleep_duration = strtol(argv[1], NULL, 10);
+
+	if (sleep_duration > 0) {
+		k_sleep(K_SECONDS(sleep_duration));
+		return 0;
+	}
+	mosh_print("sleep: duration must be 1...%ld\n", LONG_MAX);
```

Comment:
The timeout is converted into system clock ticks, of type `k_ticks_t`. That's an `uint32_t`, so it can hold the maximum value of 2^32. Our system clock is 32768 ticks per second, so the maximum timer duration is 131072 seconds. That is more than 36 hours, but just thinking about the upper value here, because `LONG_MAX` (and anything greater than 131072) doesn't work as expected, I think.

131072 seconds is more than 36 hours, so this is most likely not a real problem and I'm not sure if a check for the upper limit is needed. In my opinion, it would be fine to say that sleep duration must be greater than zero, then we would not be making false statements. 😃